### ISSUE TITLE
[Feature] Live chat support

### ISSUE THREAD

**Issue by HTSamurai** [TeamNewPipe/NewPipe#5171]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want
What I want is to be able to see live chat on NewPipe (with a way to turn it off in settings of course).

#### Is your feature request related to a problem? Please describe it
This streamer for example that I watch reads out stuff in the chat sometimes, but how could I see the chat myself? That is what this idea came from.

#### Additional context
*Two mock ups actually made from a VOD*
![IMG_20201214_160350](https://user-images.githubusercontent.com/71007160/102106013-faf5ee00-3e27-11eb-822a-e36c94ff5580.jpg)
![IMG_20201214_161726](https://user-images.githubusercontent.com/71007160/102106038-00533880-3e28-11eb-9b73-3a446311a026.jpg)
*Maybe, you could change super chat donation colours too? And maybe change the screen side it is on?*

#### How will you/everyone benefit from this feature?
People could read the live chat on NewPipe.

**Comment by HTSamurai** [comment]
"Thank you *name* for the Super chat!" she says. To which I say, "What did *name* say?"

**Comment by eladkarako** [comment]
The service is deployed to this path <code>youtube.com/live_chat</code> and displayed in the DOM container with the ID of <code>ytd-live-chat-frame</code>,  
I'm pretty sure Live Chat requires the Google unified authentication, and donation (Super Chat) are not enabled for all countries.  
Hmm.. perhaps some workaround with <code>youtube-nocookie.com</code> can be display a read-only version of chat?..

**Comment by HTSamurai** [comment]
> The service is deployed to this path `youtube.com/live_chat` and displayed in the DOM container with the ID of `ytd-live-chat-frame`,
> I'm pretty sure Live Chat requires the Google unified authentication, and donation (Super Chat) are not enabled for all countries.
> Hmm.. perhaps some workaround with `youtube-nocookie.com` might be able to display a read-only version of chat?..

I was thinking read-only anyway

**Comment by ggdorman** [comment]
I think displaying the read-only chat is a great idea. I'm not so sure about the comments overlaying the video, but having them below the video would be great.
I think this chat UI would look good:
![screen-1](https://user-images.githubusercontent.com/76399981/102721215-5ffe8780-42f1-11eb-824e-a09b5e548c09.jpg)

**Comment by HTSamurai** [comment]
> I think displaying the read-only chat is a great idea. I'm not so sure about the comments overlaying the video, but having them below the video would be great.
> I think this chat UI would look good:
> ![screen-1](https://user-images.githubusercontent.com/76399981/102721215-5ffe8780-42f1-11eb-824e-a09b5e548c09.jpg)

Or it could be optional to display over the video. I would prefer it to show over the video but I get why sone people woudn't

**Comment by ah1102** [comment]
Chat is needed in the newpipe program

**Comment by HTSamurai** [comment]
I don't think this is going anywhere. It's not had any activity. Should I close it?

**Comment by opusforlife2** [comment]
There is no time limit. Any developer who wants this feature as well could look at this issue and decide to open a PR. If you close it, that possibility may get reduced by a lot, since people usually don't browse closed issues.

**Comment by HTSamurai** [comment]
Ah ok

**Comment by iNtEgraIR2021** [comment]
I couldn't find a working parser for the live chat written in java while searching the internet.

But I found the python module [`pytchat`](https://github.com/taizan-hokuto/pytchat). This works just by http requests. So no web scraping or youtube api calling.

An example from the docs to get the live chat content as json:

```python
import pytchat
import time

chat = pytchat.create(video_id="bw-zCrtCFs0")
while chat.is_alive():
    print(chat.get().json())
    time.sleep(5)
    
    # Each chat item can also be output in JSON format.
    for c in chat.get().items:
        print(c.json())
```

So if anyone has the know-how to create a PoC in java based on this that would be great.

I'd love to do this myself but unfortunately I don't have the needed knowledge of java.

**Comment by InfinityLoop1309** [comment]
for anyone who is still interested in this feature and don't mind to use a fork: try [AnimePipe](https://github.com/InfinityLoop1309/AnimePipe), where live chat is displayed in another form - bullet comments. Chat in live replays are also supported.
Screenshot:
<img src="https://github.com/InfinityLoop1309/AnimePipe/blob/main/fastlane/metadata/android/en-US/images/phoneScreenshots/00.png?raw=true">

**Comment by 340dt7zw** [comment]
For anyone still caring about this and not mind using a fork: try [AnimePipe](https://github.com/340dt7zw/AnimePipe)

**Comment by Defog7277** [comment]
It's been four years since the creation of this Feature Request.

The ability to read Live Chat is a much needed feature. On behalf of people who like to watch streams, I would ask you to add such an option.

Is there any progress on this topic since 2020?

**Comment by swiftgeek** [comment]
Embedding `https://www.youtube.com/live_chat?is_popout=1&v=video_id_here` with some changes to visibility of elements could be an (highly inefficient) alternative, or could be worked around by using browser with that url + split screen

**Comment by joeky888** [comment]
[PipePipe](https://github.com/InfinityLoop1308/PipePipe) (fork of NewPipe) does support youtube live chat through bullet comments (danmaku subtitling).

**Comment by dezienglish81-ship-it** [comment]
Can you guys fix newpipe quality problem cause sometimes I can watch videos in the highest quality but then when I click on another video its 360p quality and I have to keep on force stopping the app after every single video just to watch high quality videos

**Comment by dezienglish81-ship-it** [comment]
when I click on YouTube channels the homepage to that YouTube channel won't load
